examples/verif-manual/vm-299.py

```diff
@@ -0,0 +1,264 @@
+"""
+VM299 Sound Diffusion in a Flat Room
+========================================================================
+
+Description:
+Sound diffusion is modeled in a flat room of size 30x30x3 m3. A sound source is placed at (2,2,1)
+with a sound power level of 1x10-2 W. The wall absorption coefficient is equal to 0.1.
+The coefficient of atmospheric attenuation is 0.01 m-1.
+
+"""
+import math
+import os
+
+# Importing the `launch_mapdl` function from the `ansys.mapdl.core` module
+from ansys.mapdl.core import launch_mapdl
+
+# Launch MAPDL with specified options
+mapdl = launch_mapdl(loglevel="WARNING", print_com=True, remove_temp_dir_on_exit=True)
+# Clear the current database
+mapdl.clear()
+
+# Run the FINISH command to exists normally from a processor
+mapdl.finish()
+
+# Set the ANSYS version
+mapdl.com("ANSYS MEDIA REL. 2022R2 (05/13/2022) REF. VERIF. MANUAL: REL. 2022R2")
+
+# Run the /VERIFY command for VM299
+mapdl.run("/VERIFY,VM299")
+
+# Set the title of the analysis
+mapdl.title("VM299 SOUND PRESSURE LEVEL IN A FLAT ROOM")
+
+"""
+The references for the analysis can be found here:
+-REFERENCE: A.BILLON,J.PICAUT,'INTRODUCING ATMOSPHERIC ATTENUATION
+WITHIN A DIFFUSION MODEL FOR ROOM-ACOUSTIC PREDICTIONS MARCH 2008.
+"""
+
+# Entering the PREP7 environment in MAPDL
+mapdl.prep7()
+
+# It is not recommended to use '/NOPR' in a normal PyMAPDL session.
+mapdl._run("/NOPR")
+
+# Constant value of PI
+pi = math.pi  # need to add "import math" at the beginning of the file
+
+# Set parameters for ROOM SIZE
+LX = 30
+LY = 30
+LZ = 3
+VOL = LX * LY * LZ
+SURF = 2 * (LX * LY + LY * LZ + LX * LZ)
+MFP = 4 * VOL / SURF
+
+# set parameters for MATERIAL PROPERTIES
+C0 = 343
+RHO = 1.21
+ROOMD = MFP * C0 / 3
+ATTN = 0.01
+ROOMDP = ROOMD / (1.0 + ATTN * MFP)
+ALPHA = 0.1
+WS = 1.0e-2
+
+# DEFINE MATERIALS
+mapdl.mp("DENS", 1, RHO)
+mapdl.mp("SONC", 1, C0)
+mapdl.tb("AFDM", 1, "", "", "ROOM")
+mapdl.tbdata(1, ROOMDP, ATTN)
+# GENERATE GEOMETRY
+H = 0.5
+mapdl.dim("A", "ARRAY", 3)
+mapdl.dim("B", "ARRAY", 3)
+mapdl.dim("C", "ARRAY", 3)
+
+mapdl.vfill("A(1)", "DATA", 0)
+mapdl.vfill("A(2)", "DATA", 2.0)
+mapdl.vfill("A(3)", "DATA", LX)
+mapdl.vfill("B(1)", "DATA", 0)
+mapdl.vfill("B(2)", "DATA", 2.0)
+mapdl.vfill("B(3)", "DATA", LY)
+mapdl.vfill("C(1)", "DATA", 0)
+mapdl.vfill("C(2)", "DATA", 2.0)
+mapdl.vfill("C(3)", "DATA", LZ)
+
+# Enter non-interactive mode
+with mapdl.non_interactive:
+    mapdl.run("*DO,I,1,2")
+    mapdl.run("*DO,J,1,2")
+    mapdl.run("*DO,K,1,2")
+    mapdl.block("A(I)", "A(I+1)", "B(J)", "B(J+1)", "C(K)", "C(K+1)")
+    mapdl.run("*ENDDO")
+    mapdl.run("*ENDDO")
+    mapdl.run("*ENDDO")
+# mapdl.aplot()
```

Comment:
You can use Python loops instead of MAPDL ones.
```suggestion
a = np.array([0, 2.0, LX]) # import numpy as np (beginning of the file)
b = np.array([0, 2.0, LY])
c = np.array([0, 2.0, LZ])

for i in range(2):
    for j in range(2):
        for k in range(2):
            mapdl.block(a[i], a[i+1], b[j], b[j+1], c[k], c[k+1])
```